cmd/ebrelayer/txs/registry.go

```diff
@@ -13,44 +13,51 @@ import (
 	bridgeregistry "github.com/Sifchain/sifnode/cmd/ebrelayer/contract/generated/artifacts/contracts/BridgeRegistry.sol"
 )
 
-// TODO: Update BridgeRegistry contract so that all bridge contract addresses can be queried
-//		in one transaction. Then refactor ContractRegistry to a map and store it under new
-//		Relayer struct.
-
 // ContractRegistry is an enum for the bridge contract types
 type ContractRegistry byte
 
 const (
-	// Valset valset contract
-	Valset ContractRegistry = iota + 1
-	// Oracle oracle contract
-	Oracle
 	// BridgeBank bridgeBank contract
-	BridgeBank
+	BridgeBank ContractRegistry = iota + 1
 	// CosmosBridge cosmosBridge contract
 	CosmosBridge
 )
 
+var bridgeBankAddress = common.HexToAddress(nullAddress)
+var cosmosBridgeAddress = common.HexToAddress(nullAddress)
+
 // String returns the event type as a string
 func (d ContractRegistry) String() string {
-	return [...]string{"valset", "oracle", "bridgebank", "cosmosbridge"}[d-1]
+	return [...]string{"bridgebank", "cosmosbridge"}[d-1]
 }
 
 // GetAddressFromBridgeRegistry queries the requested contract address from the BridgeRegistry contract
 func GetAddressFromBridgeRegistry(client *ethclient.Client, registry common.Address, target ContractRegistry,
 	sugaredLogger *zap.SugaredLogger) (common.Address, error) {
+	// Return address if already got and stored
+	switch target {
+	case BridgeBank:
+		if bridgeBankAddress.Hex() != nullAddress {
```

Comment:
I think the current code works fine, so the following idea isnt required for this PR

I would consider packing L52 - 84 into it's own function, call it `func lazyloadBridgeBankAddress()`, which loads value for bridgebankAddress and returns void, and also returns immediately if bridgebankAddress is not null. 
```
func lazyloadBridgeBankAddress(registry common.Address, client *ethclient.Client) {
    if bridgeBankAddress.Hex() != nullAddress {
        // We have already loaded bridgebank's address, no need to load again
        return nil
    }
    // Logic from L52 - 84, interact with bridgeregistry.sol to get address and memoize
}
```


Then in this switch statement, it would look like
```
switch target {
case BridgeBank:
    if err := lazyloadBridgeBank(registry); err != nil {
        return bridgeBankAddress, nil
    } else {
        return common.Address{}, err
    }
case CosmosBridge:
// Same thing
}
```

The benefit is 
- you can test the load functions separately, 
- All of the errors are inside the load function 
- The case statement is the place that returns the value for this function, so it's easier to read